astropy/io/votable/tree.py

```diff
@@ -181,10 +181,10 @@ def lookup_by_attr(self, ref, before=None):
             if element is before:
                 if getattr(element, attr, None) == ref:
                     vo_raise(
+                        KeyError,
```

Comment:
How did this ordering error not trigger in CI for the past 14 years? 😱

Reply:
My working theory is that the code has always raised (since the code assumes an exception class but falls over when it's given a string) but nothing in CI was in fact checking the actual exception class.